include/fast_float/ascii_number.h

```diff
@@ -112,7 +112,8 @@ FASTFLOAT_SIMD_RESTORE_WARNINGS
 #endif // FASTFLOAT_SSE2
 
 // dummy for compile
-template <typename UC, FASTFLOAT_ENABLE_IF(!has_simd_opt<UC>())>
+//template <typename UC, FASTFLOAT_ENABLE_IF(!has_simd_opt<UC>())>
```

Comment:
@pharago Just wondering what the compilation issue was and if it can be solved in another way. 
The enable_if is there to disable that overload when it's not needed, because of a C++ gotcha where template functions are preferred over non-template ones if the non-template one requires a conversion. It could lead to a bug later.

Reply:
> @Pharago Just wondering what the compilation issue was and if it can be solved in another way. The enable_if is there to disable that overload when it's not needed, because of a C++ gotcha where template functions are preferred over non-template ones if the non-template one requires a conversion. It could lead to a bug later.

1>c:\develop\fast_float\fast_float\include\fast_float\ascii_number.h(159): error C2665: 'fast_float::simd_read8_to_u64': none of the 2 overloads could convert all the argument types
1>c:\develop\fast_float\fast_float\include\fast_float\ascii_number.h(88): note: could be 'uint64_t fast_float::simd_read8_to_u64(const char16_t *)'
1>c:\develop\fast_float\fast_float\include\fast_float\ascii_number.h(73): note: or       'uint64_t fast_float::simd_read8_to_u64(const __m128i)'
1>c:\develop\fast_float\fast_float\include\fast_float\ascii_number.h(159): note: while trying to match the argument list '(const char *)'
1>c:\develop\fast_float\fast_float\include\fast_float\digit_comparison.h(208): note: see reference to function template instantiation 'uint32_t fast_float::parse_eight_digits_unrolled<UC>(const UC *) noexcept' being compiled

Reply:
The original code compiles unmodified using VS2022, can't say for sure what the issue is with VS2015's compiler, but removing the "has_simd_opt<UC>()" call was the only solution I could find, macro seems to expand well with a boolean as in "FASTFLOAT_ENABLE_IF( true )"
I stumbled into this issue because v140 is the compiler I used to build the new parser, to prevent the accidental inclusion of any part of the latest C++ standards

Reply:
Looks like VS 2015's implementation of template SFINAE is buggy. I added a fix that keeps the enable_if in #213 and tested it works on v140. Let me know if this works on your end.